docs/adr/0002-recipe-rewrite.md

```diff
@@ -0,0 +1,254 @@
+# 2. Recipe rewrite
```

Comment:
initial review is just on the adr - thank you for this by the way, it makes large changes like this much better :)